core-container/entry.sh

```diff
@@ -23,145 +23,133 @@ function echoerr() {
    echo "ERROR: $1" >&2
 }
 
-
-function set_provider_adapters() {
-    oldKubeconfig=$KUBECONFIG
-    KUBECONFIG=
-    echo "Waiting for space manager to be ready... this may take a while."
+function wait_kcp_ready() {
+    echo "Waiting for kcp to be ready... this may take a while."
     (
-        until [ "$(kubectl logs $(kubectl get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}') -c space-manager | grep '***READY***')" != "" ]; do
-            sleep 10
+        until [ "$(kubectl --kubeconfig $host_kubeconfig logs $(kubectl --kubeconfig $host_kubeconfig get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}') -c kcp | grep '***READY***')" != "" ]; do
+           sleep 10
         done
     )
+}
+
+function get_kcp_kubeconfig() {
+    wait_kcp_ready
+    kcp_kubeconfig_dir=/home/kubestellar/.kcp
+    kcp_kubeconfig=${kcp_kubeconfig_dir}/admin.kubeconfig
+    echo "Copying the admin.kubeconfig from kubestellar seret..."
+    mkdir -p $kcp_kubeconfig_dir
+    (
+        kubectl --kubeconfig $host_kubeconfig get secrets kubestellar -o 'go-template={{index .data "admin.kubeconfig"}}' | base64 --decode > $kcp_kubeconfig
```

Comment:
Did you know that you could alternatively extract that Secret datum as follows?

```
kubectl --kubeconfig $host_kubeconfig get secrets kubestellar -o jsonpath='{.data.admin\.kubeconfg}'
```

`kubectl wait` is your friend. The fact that it uses JSONPath should not be a barrier to using it.